cognite/client/data_classes/workflows.py

```diff
@@ -1240,7 +1272,7 @@ class WorkflowScheduledTriggerRule(WorkflowTriggerRule):
 
     _trigger_type = "schedule"
 
-    def __init__(self, cron_expression: str | None = None) -> None:
+    def __init__(self, cron_expression: str | None) -> None:
         super().__init__()
```

Comment:
These empty super-init calls look weird and may be removed by someone not looking too closely. I would advise to remove them and instead change the parent class:

```py
class WorkflowTriggerRule(CogniteObject, ABC):
    """This is the base class for all workflow trigger rules."""

    _trigger_type: ClassVar[str]

    def __init__(self) -> None:
        self.trigger_type = self._trigger_type
```

to e.g. this:
```py
class WorkflowTriggerRule(CogniteObject, ABC):
    """This is the base class for all workflow trigger rules."""

    _trigger_type: ClassVar[str]

    @property
    def trigger_type(self) -> str:
        return self._trigger_type
```

Reply:
It seems the suggestion with the @property breaks

tests/utils.py

```
signature = inspect.signature(resource_cls.__init__)
```

Is there an obvious fix? I feel I'm spending more time reverse engineering a unit test than doing something useful at the moment 🙈